sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -98,10 +98,10 @@ class RelationalGroupedDataset protected[sql](
       // Make sure all specified columns are numeric.
       colNames.map { colName =>
         val namedExpr = df.resolve(colName)
-        if (!namedExpr.dataType.isInstanceOf[NumericType]) {
+        if (!TypeCollection.NumericAndInterval.acceptsType(namedExpr.dataType)) {
```

Comment:
Can you update the comment to make it more general one?
https://github.com/apache/spark/blob/c2d513f8e9f5d96dae7990325617013bcc92fe5d/sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala#L82

Reply:
thanks for your suggestion. Please check 。